ffenv/env.go

```diff
@@ -0,0 +1,57 @@
+package ffenv
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+// Parser is a parser for .env file format: flag=value. Each
+// line is tokenized as a single key/value pair.
+func Parser(r io.Reader, set func(name, value string) error) error {
+	return parse("", r, set)
+}
+
+func parse(prefix string, r io.Reader, set func(name, value string) error) error {
+	s := bufio.NewScanner(r)
+	for s.Scan() {
+		line := strings.TrimSpace(s.Text())
+		if line == "" {
+			continue // skip empties
+		}
+
+		if line[0] == '#' {
+			continue // skip comments
+		}
+
+		line = strings.TrimPrefix(line, prefix+"_")
+		var (
+			name  string
+			value string
+			index = strings.IndexRune(line, '=')
+		)
+		if index < 0 {
+			name, value = line, "true" // boolean option
```

Comment:
I think in this parser we don't want to interpret lines of the form

```
FOO_BAR
```

as setting boolean flags to true. I think they should probably be errors.

Reply:
I think that's fair.

Reply:
this is fixed in 23a2085